frontend/catalyst/utils/toml.py

```diff
@@ -261,123 +229,52 @@ def get_operation_properties(config_props: dict) -> OperationProperties:
-        "QubitUnitary",
-        "ControlledQubitUnitary",
-        "Toffoli",
-    ]
-
-    supports_controlled = check_quantum_control_flag(config)
-    if supports_controlled:
-        # Add ControlledQubitUnitary as a controlled version of QubitUnitary
-        if "QubitUnitary" in native_gate_props:
-            native_gate_props["ControlledQubitUnitary"] = OperationProperties(
-                invertible=False, controllable=False, differentiable=True
-            )
-        # By default, enable the `C(gate)` version for most `gates`.
-        for op, props in native_gate_props.items():
-            props.controllable = op not in gates_to_be_decomposed_if_controlled
-
-    supports_adjoint = get_compilation_flag(config, "quantum_adjoint")
-    if supports_adjoint:
-        # Makr all gates as invertibles
-        for props in native_gate_props.values():
-            props.invertible = True
-
-    # Mark all gates as differentiable
-    for props in native_gate_props.values():
-        props.differentiable = True
-    # Mark all observables as differentiable
-    for props in observable_props.values():
-        props.differentiable = True
-
-    # For toml schema 1 configs, the following condition is possible: (1) `QubitUnitary` gate is
-    # supported, (2) native quantum control flag is enabled and (3) `ControlledQubitUnitary` is
-    # listed in either matrix or decomposable sections. This is a contradiction, because
-    # condition (1) means that `ControlledQubitUnitary` is also in the native set. We solve it
-    # here by applying a fixup.
-    # TODO: remove after PR #642 is merged in lightning
-    if "ControlledQubitUnitary" in native_gate_props:  # pragma: nocover
-        if "ControlledQubitUnitary" in matrix_decomp_props:
-            matrix_decomp_props.pop("ControlledQubitUnitary")
-        if "ControlledQubitUnitary" in decomp_props:
-            decomp_props.pop("ControlledQubitUnitary")
-
-    # Fix a bug in device toml schema 1
-    if "ControlledPhaseShift" in native_gate_props:  # pragma: nocover
-        if "ControlledPhaseShift" in matrix_decomp_props:
-            matrix_decomp_props.pop("ControlledPhaseShift")
-        if "ControlledPhaseShift" in decomp_props:
-            decomp_props.pop("ControlledPhaseShift")
+def _validate_schema_version(schema):
+    """Pass on supported schema versions. Otherwise, raise an error"""
+
+    # TODO: remove deprecation message when appropriate
+    assert schema not in ALL_DEPRECATED_SCHEMAS, f"Schema {schema} is deprecated"
+
+    assert schema in ALL_SUPPORTED_SCHEMAS, f"Unsupported config schema {schema}"
 
 
 def load_device_capabilities(
-    config: TOMLDocument, program_features: ProgramFeatures, device_name: str
+    config: TOMLDocument, program_features: ProgramFeatures
 ) -> DeviceCapabilities:
     """Load device capabilities from device config"""
 
-    schema = int(config["schema"])
+    _validate_schema_version(int(config["schema"]))
```

Comment:
Might be more of an individual style question, but personally I'm not crazy about placing underscores everywhere, especially in internal code. I do find them useful though in distinguishing between "public" and "private" methods in a user facing class though for example.

Reply:
Before the PR, there were multiple functions with repeated code for validating the schema, probably because they were (unnecessarily) public. They even were included in the tests. I have then centralized the validation in one single point, after which all those other functions are invoked. I made them private with the underscore to prevent them from being called outside the module, as they are only needed inside it.